omegaconf/_utils.py

```diff
@@ -501,21 +467,18 @@ def is_primitive_type(type_: Any) -> bool:
 
 def _is_interpolation(v: Any) -> bool:
     if isinstance(v, str):
-        ret = get_value_kind(v) in (
-            ValueKind.INTERPOLATION,
-            ValueKind.STR_INTERPOLATION,
-        )
-        assert isinstance(ret, bool)
-        return ret
+        return bool(get_value_kind(v) == ValueKind.INTERPOLATION)
     return False
 
 
 def _get_value(value: Any) -> Any:
     from .base import Container
     from .nodes import ValueNode
 
-    if isinstance(value, Container) and value._is_none():
-        return None
+    if isinstance(value, Container) and (
+        value._is_none() or value._is_interpolation() or value._is_missing()
+    ):
+        return value._value()
```

Comment:
curious about this change. Anything changed that requires it?

Reply:
So, this is a tricky one to explain, because it's tightly related to changes in this PR, and since it's something I've been working on for a while I don't remember all the details -- I had to investigate to answer this question. Details below, but TLDR: it makes the code simpler.

First, you may notice that I removed the first three lines in the excerpt below from `get_value_kind()`:

```python
    if isinstance(value, Container):	
        if value._is_interpolation() or value._is_missing():	
            value = value._value()	

    value = _get_value(value)
```

If I change it back (moving back this check to `get_value_kind()` and removing it from `_get_value()`, there is an infinite recursion that comes up due to trying to parse a DIctConfig as an interpolation (while we should parse the underlying interpolation string, not the DictConfig itself).

This infinite recursion can be fixed by adding to `_resolve_interpolation()`
```python
        if isinstance(value, Container):	
            value = value._value()
```
just before calling `parse_tree = parse(_get_value(value))`, and then everything would work fine.

So essentially, there are two places where before calling `_get_value()` on a container, with the previous implementation of `_get_value()` we had to do this check and access the underlying value instead. It seemed more sensible to me to move this check to `_get_value()` so as to simplify the rest of the code. I couldn't see any drawback, and all tests succeeded.

Note that I added tests for this new behavior of `_get_value()`, so by reverting the code as I described above, when I said "everything would work fine", it's not entirely true: my new tests would fail :)

Reply:
Yeah sorry it was hard to explain, the gist of it is really in this paragraph:

> So essentially, there are two places where before calling _get_value() on a container, with the previous implementation of _get_value() we had to do this check and access the underlying value instead. It seemed more sensible to me to move this check to _get_value() so as to simplify the rest of the code

=> resolving the conversation